parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -578,14 +608,8 @@ public ColumnChunkPageReader readAllPages() throws IOException {
             if (dictionaryPage != null) {
               throw new ParquetDecodingException("more than one dictionary page in column " + descriptor.col);
             }
-          DictionaryPageHeader dicHeader = pageHeader.getDictionary_page_header();
-          dictionaryPage =
-                new DictionaryPage(
-                    this.readAsBytesInput(compressedPageSize),
-                    uncompressedPageSize,
-                    dicHeader.getNum_values(),
-                    converter.getEncoding(dicHeader.getEncoding())
-                    );
+
+            dictionaryPage = readDictionary(configuration, pageHeader, descriptor.metadata, f);
```

Comment:
Yeah, I think I mentioned before that this probably doesn't work.  We had some discussion about trying to consolidate the dictionary read logic so it can be exposed and used by other tools.

We can probably rework this to do both.